### PR TITLE
chore: add HasValidateBasic compiler assertions to msgs

### PR DESCRIPTION
**Pull request by trinitys7**
## Description

Add the HasValidateBasic interface as a compiler assertion to each of the ibc-go tx msgs.

closes: #4665

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]

**Review by DimitrisJim — APPROVED** [review]
lgtm, go fmt is needed

**Review by charleenfei — APPROVED** [review]

**Comment by colin-axner** [comment]
Hi @trinitys7, looks like the linter needs to be fixed and the branch will need to be kept up to date with main before merge. We will be happy to do this for you if you could allow us to push to your branch (usually there's something you can click on the right hand side of this pr)

**Comment by trinitys7** [comment]
@colin-axner, I made this PR from my organization fork, which doesn't allow maintainers to make edits. I will close this PR and create another one from my personal fork instead.

**Pull request closed without merging**
